Add more detailed docs for ZenML Pro (#3065)
zenml-io/zenml: docs/book/getting-started/zenml-pro/roles.md

## Tenant-Level Roles

Tenant roles determine a user's permissions within a specific ZenML tenant. There are predefined roles available, and you can also create custom roles for more granular control.

![Image showing the tenant roles](../../.gitbook/assets/role_page.png)

### Predefined Tenant Roles

1. **Admin**
   - Full control over the tenant
   - Can create, read, update, and delete all resources

![Image showing the admin role](../../.gitbook/assets/admin_role.png)

2. **Editor**
   - Can create, read, and share resources
   - Cannot modify or delete existing resources

3. **Viewer**
   - Read-only access to all resources and information

### Custom Roles

Custom roles allow you to define specific permissions for users or groups. To create a custom role
for a tenant:

1. Go to the tenant settings page

![Image showing the tenant settings page](../../.gitbook/assets/custom_role_settings_page.png)

2. Click on "Roles" in the left sidebar and Select "Add Custom Role"

![Image showing the add custom role page](../../.gitbook/assets/tenant_roles_page.png)

> 3. Provide a name and description for the role. Choose a base role to inherit permissions from

Reviewer: ```suggestion
3. Provide a name and description for the role. Choose a base role from which to inherit permissions
```

- Artifacts
- Models
- Model Versions
- Pipelines
- Runs
- Stacks
- Components
- Secrets
- Service Connectors

For each resource, you can define the following permissions:

- Create
- Read
- Update
- Delete
- Share

You can then assign this role to a user or a team on the "Members" page.

#### Managing permissions for roles

To manage permissions for a role:

1. Go to the Roles page in tenant settings
2. Select the role you want to modify
3. Click on "Edit Permissions"
4. Adjust permissions for each resource type as needed

![Assign Permissions](../../.gitbook/assets/assign_permissions.png)

## Sharing individual resources

While roles define permission on broad resource groups, users can also share individual resources
through the dashboard like this:

![Share dialog](../../.gitbook/assets/share_dialog.png)

## Best Practices

1. **Least Privilege**: Assign the minimum necessary permissions to each role.
2. **Regular Audits**: Periodically review and update role assignments and permissions.
3. **Use Custom Roles**: Create custom roles for teams or projects with specific needs.
4. **Document Roles**: Maintain documentation of your custom roles and their intended use.

> By leveraging ZenML Pro's role-based access control, you can ensure that your team members have the right level of access to resources, maintaining security while enabling collaboration across your MLOps projects.<!-- For scarf -->

Reviewer: ```suggestion
By leveraging ZenML Pro's role-based access control, you can ensure that your team members have the right level of access to resources, maintaining security while enabling collaboration across your MLOps projects.
<!-- For scarf -->
```

## Organization Roles

At the organization level, you can assign the following roles to people.

![Image showing the organization roles](../../.gitbook/assets/org_members.png)

- **Org admin**: This role has full control over the organization. They can add
  members, create and update tenants, adjust the billing information and assign roles.
- **Org editor**: This role can manage tenants and members but is not allowed to
  access the subscription information or delete the organization.
- **Org viewer**: This role can view the tenants within the organization with only
  view permissions.

Some points to note:
- In addition to adding organization roles, you might also want to add tenant roles for people who you want to have access to a specific tenant.
- An organization admin can add themselves to a tenant with any tenant role they desire.
- However, an organization editor and viewer cannot add themselves to existing tenants that they are not a part of. They won't be able to view such tenants in the organization either.

> You can also send invites to new members with a specific role.

Reviewer: ```suggestion
```

You said this already above.

> ![Image showing the invite modal](../../.gitbook/assets/add_org_members.png)

Reviewer: ```suggestion
![Screenshot showing the invite modal](../../.gitbook/assets/add_org_members.png)
```

## Tenant Roles

> Once you have added people to your organization, they can start interacting with the tenants in it, respecting the role they have been assigned. Each user needs to also have a tenant role to be able to perform any actions inside a ZenML tenant. You can choose from a selection of predefined roles or create your own custom role.

Reviewer: ```suggestion
Once you have added people to your organization, they can start interacting with the tenants in it, taking into account the role they have been assigned. Each user needs to also have a tenant role to be able to perform any actions inside a ZenML tenant. You can choose from a selection of predefined roles or create your own custom role.
```
